README.md
亨元金融 ( Hengyuan Finance ) ![Build Passing](http://img.shields.io/travis/joyent/node/v0.6.svg)
=================================================================================================
后台管理系统前端框架 ( Backstage Management System Front-End Framework )

CHANGELOG.md
###针对按钮模块的初始化提交（PM 2014.04.24.16.24）###
-----------------------------------------------------

[里程碑 " 0.1.0-alpha " 镜像文件更新/提交记录](http://goo.gl/FiI3hn)。

此次提交共涉及 **56** 个文件，查看此次具体[提交记录](http://goo.gl/VySrc6), 点击该页面下 " Show Diff Status " 可以快速查看此次提交涉及哪些个文件。

[预览](http://goo.gl/uw5wg8)



###针对表格模块的初始化提交（PM 2014.04.17.17.17）###
-----------------------------------------------------

此次提交共涉及 **11** 个文件，查看此次具体[提交记录](http://goo.gl/MWlON7), 点击该页面下 " Show Diff Status " 可以快速查看此次提交涉及哪些个文件。

[预览](http://goo.gl/uw5wg8)



###针对项目及母版页的初始化提交（PM 2014.04.16.17.28）###
---------------------------------------------------------

此次提交共涉及 **103** 个文件，查看此次具体[提交记录](http://goo.gl/3LdKPA), 点击该页面下 " Show Diff Status " 可以快速查看此次提交涉及哪些个文件。

[预览](http://goo.gl/uw5wg8)

public/ajax/libs/js/bms_front_end_framework/app/master.main.js
/**
 * Statement: ...//TODO: Write statement.
 * 
 * Describe:     ...//TODO: Check description.
 * 
 * Further Changes, Comments: ...//TODO: Give a further changes and comments link.
 * 
 * Javascript Design Pattern (Code Management):    ...//TODO: Cehck design pattern.
 *     
 *     Namespacing Patterns, Immediately-invoked Function Expressions (IIFE)s
 *     
 *     Modules Patterns, Revealing Module Pattern
 *     
 *     Modules Patterns, AMD modules
 *     
 * Docs: ...//TODO: Give a link about project documents.
 * 
 * Original Author: Shen Weizhong ( Tony Stark ).
 * 
 * Thanks: ...//TODO: If there are some contributors, just mark them.
 * 
 * Version: 0.1.0-alpha
 * 
 * Creation Date: 2014.04.15 16:46 ( Tony ).
 * 
 * Last Update: 2014.04.15 16:52 ( Tony ).    ...//TODO: Update the 'Last Update'.
 * 
 * Music ( Custom ): Countdown (feat. Makj).mp3    ...//TODO: If you are listenning some music, just write the name of songs.
 * 
 * License: ...//TODO: Give a license.
 * 
 * Copyright: ...//TODO: Give a copyright.
 */

(function (require) {
	
	var fn, _AMD;
	
	fn = function (require) {
		
		var SJ, srl, easing, modernizr, tlns, extend, _mod, mute, evtName;
		
		SJ = require('jquery');
		
		srl = require('srl');

		modernizr = require('modernizr');
		
		SJ(function ($) {

			tlns = tlns || {}; //top-level namespace

			modernizr.touch ? evtName = 'touchstart' : evtName = 'click';
			
			// TODO: Import basic business logic script here.

			$('a[href=#]').on('click', function (e) {

				e.stopPropagation();

				e.preventDefault();

			});

			$(':root').studioScroll({zindex: 50, cursorborder: 0, cursorborderradius: 0});

			$('aside > ul.accordian').studioScroll({zindex: 50, cursorborder: 0, cursorborderradius: 0});

			$('.accordian').find('.slider-header.current').next().slideDown();

			$('.slider-header').on(evtName, function () {

				if( $(this).next().is(':hidden') ) { 

					$('.slider-header').removeClass('current').next().slideUp(); 

					$(this).toggleClass('current').next().slideDown();

				}

			});

			var choosenItem = $('.slider-content').children('li.choosen');

			choosenItem.data('asd', 0);

			$('.slider-content').find('a').on('mouseover', function (e) {

				var _this = $(this),

					_parent = _this.parent();

				if (!_parent.hasClass('choosen')) {

					_parent.addClass('choosen');

				}

			}).on('mouseleave', function (e) {

				var _this = $(this),

					_parent = _this.parent();

				if (_parent.data('asd') !== 0) {

					_parent.removeClass('choosen');

				}

			});

			$('.slider-content').find('a').on(evtName, function () {

				var _this = $(this),

					_parent = _this.parent();

				$('.accordian').find('li.choosen').removeClass('choosen');

				choosenItem.data('asd', '');

				_parent.data('asd', 0);

				choosenItem = _parent;

				choosenItem.addClass('choosen');

			});
			
		});
		
	};
	
	_AMD = (function (_register, _module) {
		
		var hasDefine, registryProfile;
		
		hasDefine = typeof define === "function" && define.amd;
		
		registryProfile = function () {
			
			hasDefine ? define(_module) : console.error('Sorry! There is no "define" object.');
			
		};
		
		return {
			
			init: registryProfile
			
		};
		
	}(_AMD || {}, fn)).init();
	
} (require));

public/ajax/libs/js/bms_front_end_framework/common.js
/**
 * Statement: ...//TODO: Write statement.
 * 
 * Describe: Global configuration of requirejs.    ...//TODO: Check description.
 * 
 * Further Changes, Comments: ...//TODO: Give a further changes and comments link.
 * 
 * Javascript Design Pattern (Code Management):    ...//TODO: Cehck design pattern.
 *     
 *     Namespacing Patterns, Immediately-invoked Function Expressions (IIFE)s
 *     
 *     Namespacing Patterns, Nested namespacing
 *     
 *     Modules Patterns, Object literal notation
 *     
 *     Modules Patterns, Revealing Module Pattern
 *     
 *     Modules Patterns, AMD modules
 *     
 * Docs: ...//TODO: Give a link about project documents.
 * 
 * Original Author: Shen Weizhong ( Tony Stark ).
 * 
 * Thanks: ...//TODO: If there are some contributors, just mark them.
 * 
 * Version: 0.1.0-alpha
 * 
 * Creation Date: 2014.04.15 16:46 ( Tony ).
 * 
 * Last Update: 2014.04.15 16:50 ( Tony ).    ...//TODO: Update the 'Last Update'.
 * 
 * Music ( Custom ): Countdown (feat. Makj).mp3    ...//TODO: If you are listenning some music, just write the name of songs.
 * 
 * License: ...//TODO: Give a license.
 * 
 * Copyright: ...//TODO: Give a copyright.
 */

(function (global, document, requirejs, console, undefined) {
	
	"use strict";
	
	var fn, _AMD;
	
	fn = function () {
		
		var boot, baseUrl, cdnjsUrl, jq1x, jq2x;
		
		boot = boot || {};
		
		/* Begin | Options */
			
			baseUrl = '../public/ajax/libs/js';
			
			cdnjsUrl = '//resource.fenqimall.com/ajax/libs/js';
			
			jq1x = cdnjsUrl + '/jquery/1.11.0/jquery.min';
			
			jq2x = cdnjsUrl + '/jquery/2.1.0/jquery.min';
			
		/* End | Options */
		
		boot.getAgent = function () {
			
			return navigator.userAgent.toLowerCase();
			
		};
		
		boot.isIE = function(userAgent) {
			
			var agent = userAgent || this.getAgent();
			
			return !!agent.match(/msie/i);
			
		};
		
		boot.isGteIE9 = function(userAgent) {
			
			var agent = userAgent || this.getAgent(),
				
				match = agent.match(/msie\D*([\.\d]*)/i),
				
				version = 0;
			
			if (match && match[1]) {
				
				version = match[1];
				
			}
			
			return version >= 9;
			
		};
		
		boot.req = function (jquery) {
			
			requirejs.config({
				
				baseUrl: baseUrl,
				
				enforceDefine: true,
				
				paths: {
					
					modernizr: cdnjsUrl + '/modernizr_amd/modernizr.min',
					
					jqPrivate: cdnjsUrl + '/jquery_private/jquery.private.min',
					
					jquery: jquery,
					
					srl: cdnjsUrl + '/jquery_scroll/0.1.0/jquery.scroll.min',
					
					easing: cdnjsUrl + '/jquery_easing/1.3/jquery.easing',
					
					cookie: cdnjsUrl + '/jquery_cookie/1.3.1/jquery.cookie.min'
					
				},
				
				waitSeconds: 120,
				
				map: {
					
					'*': {
						
						'jquery': 'jqPrivate'
						
					},
					
					'jqPrivate': {
						
						'jquery': 'jquery'
						
					}
					
				},
				
				skipDataMain: true
				
			});
			
		};
		
		boot.judgement = function (opts) {
			
			if (this.isIE()) {
				
				this.isGteIE9() ? this.req(opts.jq2x) : this.req(opts.jq1x);
				
			} else {
				
				this.req(opts.jq2x);
				
			}
			
		};
		
		boot.judgement({
			
			jq1x: jq1x,
			
			jq2x: jq2x
			
		});
		
	};
	
	_AMD = (function (_register, _module) {
		
		var hasDefine, registryProfile;
		
		hasDefine = typeof define === "function" && define.amd;
		
		registryProfile = function () {
			
			hasDefine ? define(_module) : console.error('Sorry! There is no "define" object.');
			
		};
		
		return {
			
			init: registryProfile
			
		};
		
	}(_AMD || {}, fn)).init();
	
} (window, document, requirejs, (typeof console !== 'undefined' ? console : undefined)));